test/e2e/framework/kcp.go

```diff
@@ -326,11 +326,20 @@ func (c *kcpServer) waitForEndpoint(client *rest.RESTClient, endpoint string) {
 			lastMsg = fmt.Sprintf("error contacting %s: %v", endpoint, err)
 		}
 	}, 100*time.Millisecond)
-	if !succeeded {
+	if !succeeded && !isDone(c.ctx) {
```

Comment:
Technically if we failed before the context was cancelled but executed this line after, this would eat the error. Not sure we can do better, though?